File: packages/c#/Runner.cs
using System.Collections.Generic;
using Com.CodeGame.CodeBall2018.DevKit.CSharpCgdk.Model;

namespace Com.CodeGame.CodeBall2018.DevKit.CSharpCgdk
{
    public sealed class Runner
    {
        private readonly RemoteProcessClient remoteProcessClient;
        private readonly string token;
        public static void Main(string[] args)
        {
            new Runner(args.Length == 3 ? args : new[] { "127.0.0.1", "31001", "0000000000000000" }).Run();
        }

        private Runner(IReadOnlyList<string> args)
        {
            remoteProcessClient = new RemoteProcessClient(args[0], int.Parse(args[1]));
            token = args[2];
        }

        public void Run()
        {
            IStrategy strategy = new MyStrategy();
            IDictionary<int, Action> actions = new Dictionary<int, Action>();
            Game game;
            remoteProcessClient.WriteToken(token);
            Rules rules = remoteProcessClient.ReadRules();
            while ((game = remoteProcessClient.ReadGame()) != null)
            {
                actions.Clear();
                foreach (var robot in game.robots)
                {
                    if (robot.is_teammate)
                    {
                        Action action = new Action();
                        strategy.Act(robot, rules, game, action);
                        actions.Add(robot.id, action);
                    }
                }
                remoteProcessClient.Write(actions);
            }
        }
    }
}

File: packages/scala/src/main/scala/Runner.scala
import model._

object Runner extends App {
  override def main(args : Array[String]): Unit = {
    val newArgs = if (args.length == 3) args else Array( "127.0.0.1", "31001", "0000000000000000" )
    new Runner(newArgs).run()
  }
}

class Runner(args : Array[String])  {
  val remoteProcessClient = new RemoteProcessClient(args(0), args(1).toInt)
  val token = args(2)

  def run(): Unit = {
    val strategy = new MyStrategy()
    var actions: Map[Int, Action] = Map()
    remoteProcessClient.writeToken(token)
    val rules: Rules = remoteProcessClient.readRules()
    var game: Game = remoteProcessClient.readGame()
    while (game != null) {
      actions = Map()
      for (robot <- game.robots) {
        if (robot.is_teammate) {
          val action = strategy.act(robot, rules, game)
          actions = actions.updated(robot.id, action)
        }
      }
      remoteProcessClient.write(actions)
      game = remoteProcessClient.readGame()
    }
  }
}
